client/app/bundles/course/assessment/submission/components/ProgressPanel.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { injectIntl, intlShape } from 'react-intl';
+import { Card, CardHeader, CardText } from 'material-ui/Card';
+import Paper from 'material-ui/Paper';
+import { Table, TableBody, TableRow, TableRowColumn } from 'material-ui/Table';
+import { red100, yellow100, grey100, green100, blue100 } from 'material-ui/styles/colors';
+import WarningIcon from 'material-ui/svg-icons/alert/warning';
+
+import { formatDateTime } from '../utils';
+import { submissionShape } from '../propTypes';
+import translations from '../translations';
+import { workflowStates } from '../constants';
+
+const styles = {
+  header: {
+    attempting: {
+      backgroundColor: yellow100,
+    },
+    submitted: {
+      backgroundColor: grey100,
+    },
+    graded: {
+      backgroundColor: blue100,
+    },
+    published: {
+      backgroundColor: green100,
+    },
+  },
+  warningIcon: {
+    display: 'inline-block',
+    verticalAlign: 'middle',
+  },
+  table: {
+    maxWidth: 600,
+  },
+};
+
+class ProgressPanel extends Component {
+
+  renderLateWarning() {
+    const { intl } = this.props;
+    return (
+      <Paper style={{ backgroundColor: red100, padding: 10 }}>
+        <WarningIcon style={styles.warningIcon} />
+        <span>{intl.formatMessage(translations.lateSubmission)}</span>
+      </Paper>
+    );
+  }
+
+  renderTimes() {
+    const { intl } = this.props;
+    const { submittedAt } = this.props.submission;
+    return (
+      <Table selectable={false} style={styles.table}>
+        <TableBody displayRowCheckbox={false}>
+          <TableRow>
+            <TableRowColumn>{intl.formatMessage(translations.submittedAt)}</TableRowColumn>
+            <TableRowColumn>{formatDateTime(submittedAt)}</TableRowColumn>
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+  }
+
+  render() {
+    const { intl } = this.props;
+    const { late, submitter, workflowState } = this.props.submission;
+    const title = intl.formatMessage(translations[workflowState]);
+    return (
+      <Card>
+        <CardHeader
+          title={intl.formatMessage(translations.submissionBy, { submitter })}
+          subtitle={title}
+          style={styles.header[workflowState]}
+        />
+        <CardText>
```

Comment:
There is a lot of unnecessary spacing between the header and "Submitted At" when the submission is not late. How about moving `<CardText>` into the conditional?:

```
{late && workflowState !== workflowStates.Submitted ? 
<CardText>{this.renderLateWarning()}</CardText> : 
null}
```